hicommonwealth/commonwealth
# Only show skeleton once on discussion page

## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/4852

## Description of Changes
Now showing the skeleton loader once on the discussions page

## "How We Fixed It"
By using a ref for loader and only showing skeleton once.

## Test Plan

1. Visit the discussions page
2. Wait for 3 mins or reduce the timer [here in fetchThreads](https://github.com/hicommonwealth/commonwealth/blob/fa85edab73702aa7e513bcc3ed36e0008960cfe0/packages/commonwealth/client/scripts/state/api/threads/fetchThreads.ts#L12) to something like 3 seconds ex: `const THREADS_STALE_TIME = 3000`
3. Now verify that the skeleton does not show after 3 seconds

## Deployment Plan
N/A

## Other Considerations
Do we want to keep displaying old threads when new ones are fetching after the cache/stale timer? if we do then I think we should come up with a temporary memoization/state pattern to handle this case everywhere
